Compile options diagnostics not displayed

Compiler options error messages are not logged to the terminal in current TSC nightly builds.

STR:

1. With typescript 1.6.0-dev.20150725, run `tsc -m commonjs -t es6 test.ts --noEmitOnError` where test.ts is an empty file

Expected Result: Error message about combining commonjs with es6 (`error TS1204: Cannot compile modules into 'commonjs', 'amd', 'system' or 'umd' when targeting 'ES6' or higher.`)

Actual result: No error, but program exits with status 1

In `compiler/program.ts`, `emitWorker()` gets a list of pre-emit diagnostics but then doesn't return them:

````
function emitWorker(program: Program, sourceFile: SourceFile, writeFileCallback: WriteFileCallback, cancellationToken: CancellationToken): EmitResult {
            // If the noEmitOnError flag is set, then check if we have any errors so far.  If so,
            // immediately bail out.  Note that we pass 'undefined' for 'sourceFile' so that we
            // get any preEmit diagnostics, not just the ones
            if (options.noEmitOnError && getPreEmitDiagnostics(program, /*sourceFile:*/ undefined, cancellationToken).length > 0) {
                return { diagnostics: [], sourceMaps: undefined, emitSkipped: true };
            }
````

Changing that to:
````
// If the noEmitOnError flag is set, then check if we have any errors so far.  If so,
            // immediately bail out.  Note that we pass 'undefined' for 'sourceFile' so that we
            // get any preEmit diagnostics, not just the ones
			let preEmitDiagnostics = getPreEmitDiagnostics(program, /*sourceFile:*/ undefined, cancellationToken);
            if (options.noEmitOnError && .length > 0) {
                return { diagnostics: preEmitDiagnostics, sourceMaps: undefined, emitSkipped: true };
            }
````
Results in the diagnostics being reported. Is this the correct fix?

--- a/src/compiler/tsc.ts
+++ b/src/compiler/tsc.ts
@@ -363,7 +363,7 @@ namespace ts {
             // If we didn't have any syntactic errors, then also try getting the global and
             // semantic errors.
             if (diagnostics.length === 0) {
-                diagnostics = program.getGlobalDiagnostics();
+                diagnostics = program.getOptionsDiagnostics().concat(program.getGlobalDiagnostics());
 
                 if (diagnostics.length === 0) {
                     diagnostics = program.getSemanticDiagnostics();

